flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/operations/SqlToOperationConverter.java

```diff
@@ -542,6 +551,60 @@ private Operation convertAlterTableReset(
         return new AlterTableOptionsOperation(tableIdentifier, oldTable.copy(newOptions));
     }
 
+    private Operation convertAlterTableCompact(
+            ObjectIdentifier tableIdentifier,
+            ResolvedCatalogTable resolvedCatalogTable,
+            SqlAlterTableCompact alterTableCompact) {
+        Catalog catalog =
+                catalogManager
+                        .getCatalog(tableIdentifier.getCatalogName())
+                        .orElseThrow(
+                                () ->
+                                        new ValidationException(
+                                                String.format(
+                                                        "Catalog %s not exists",
+                                                        tableIdentifier.getCatalogName())));
+        if (ManagedTableListener.isManagedTable(catalog, resolvedCatalogTable)) {
+            LinkedHashMap<String, String> partitionKVs = alterTableCompact.getPartitionKVs();
+            CatalogPartitionSpec partitionSpec = null;
+            if (partitionKVs != null) {
+                List<String> orderedPartitionKeys = resolvedCatalogTable.getPartitionKeys();
+                int index = 0;
+                String exMsg =
+                        orderedPartitionKeys.isEmpty()
```

Comment:
I thought about it, maybe it doesn't have to be ordered, just hit the specified partition is OK.
We can just do something like `DynamicSinkUtils.validatePartitioning`